### ISSUE TITLE
Text doesn't changes in codeglance window

### ISSUE THREAD

**Issue by Losiara** [olmokramer/atom-minimap-codeglance#1]
0.1.4 
In my case text doesn't move. Only window.

**Comment by olmokramer** [comment]
This has been an issue during development, not sure why it would happen again... Did you try to restart Atom?

**Comment by Losiara** [comment]
Yes and reinstalled package.

**Comment by Losiara** [comment]
alignWithCursor: (offset) ->
    return if @position isnt 'cursor'
    translateY = offset - @clientHeight / 2
    translateY = Math.max @getMinTranslateY(), translateY
    translateY = Math.min @getMaxTranslateY(), translateY
    requestAnimationFrame =>
      @style.transform = "translateY(#{translateY}px)"

requestAnimationFrame seems broken.

**Comment by olmokramer** [comment]
What version of Atom are you on?

**Comment by olmokramer** [comment]
That piece of code is to move the codeglance window, not the text inside it...

**Comment by Alchiadus** [comment]
Preview:
![atom-minimap-codeglance-first-lines-only](https://cloud.githubusercontent.com/assets/576095/8543397/c90e236c-249c-11e5-8f7c-26b3bc9d4618.gif)
Using Atom 1.0 on Windows 10 (build 10162).

I restarted Atom, tried to disable/enable the package (and restart) and tried to disable/enable the plugin in the settings of Minimap (and restart), but sadly none had any success.

**Comment by Losiara** [comment]
Atom ver 1.0 Win 7 x64
Exactly the same as in prev comment

**Comment by olmokramer** [comment]
What happens if you replace lines 48-49 with `@editor.displayBuffer.setHeight @clientHeight`?

**Comment by Alchiadus** [comment]
Same behaviour as before for me.

**Comment by Losiara** [comment]
Also

**Comment by Losiara** [comment]
Firstly for one moment it scrolls properly than it moves to first lines.

UPD also linenumbers are not displayed

**Comment by olmokramer** [comment]
How about `@editorView.component.presenter.setExplicitHeight @clientHeight` instead of lines 48-49? Or a combination of both? Or with just lines 48-49 removed?

**Comment by Losiara** [comment]
It scrolls right but the same moment it returns back to the begining.

**Comment by olmokramer** [comment]
Ok, I think maybe I found a solution:
1. remove line `48-49` of `lib/codeglance-view.coffee`
2. add the following at the end of `styles/minimap-codeglance.atom-text-editor.less`

``` less
minimap-codeglance atom-text-editor {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
}
```

This introduces another issue, where the codeglance view will have a height of `0` the first time it's visible, but I think I can fix that as well.

**Comment by Alchiadus** [comment]
That fixed it (with the issue you already mentioned). :+1:

**Comment by Losiara** [comment]
Scrolling Ok but line numbers steel invisible.
UPD. It was my mistake, not the issue.

**Comment by olmokramer** [comment]
Ok, I've pushed a new version. Didn't publish it yet, though. Could you uninstall the package and `git clone` this repo into `~/.atom/packages` and check if everything works?

**Comment by olmokramer** [comment]
About the line numbers, do you have the `showGutter` setting enabled? If yes, and the gutter still isn't working, could you please open a new issue for that?

**Comment by olmokramer** [comment]
Oops, I renamed the `showGutter` setting to `showLineNumbers` in the latest commit

**Comment by Alchiadus** [comment]
Last version works fine for me, but seems to introduce a visual issue. Every time codeglance is displayed (i.e. you move your mouse from outside the minimap on the minimap) the top and bottom border of codeglance flicker white before the correct color is applied. Scrolling after that works fine though.

**Comment by olmokramer** [comment]
Thanks. If you change line 37 to `@style.opacity = if @clientHeight is 0 then 0 else 1`, does that stop it from happening?

**Comment by Alchiadus** [comment]
Yes, that fixes it. I also caught something I did not notice earlier, which only occurs the very first time codeglance is displayed after an Atom (re)start. if you move your mouse on the minimap with no vertical movement, codeglance will not scroll to the position your mouse is hovering over the minimap, but rather display the first `#numberOfLines` lines in the file. If you'd like me to make a separate issue please let me know.

**Comment by olmokramer** [comment]
Yes, that might happen the very first time it is shown. I don't understand exactly why it does that - the editor is quite complex - but I think it's something to do with line measurements. The very first time you hover the minimap is also the very first time the editor is attached to the dom, so before then it can't do its measurements I think. And apparently that takes some time. I tried to work around this by attaching the editor earlier, but so far those attempts have failed. I noticed that in general all works well after _moving_ over the minimap for about 300ms, so I don't worry about it too much.
